tests/StatusBar.test.ts

```diff
@@ -29,10 +33,31 @@ describe('StatusBar', () => {
     statusBarItem.text = ''
   })
 
-  describe('registerCommand()', () => {
-    it('should set statusBarItem command', () => {
-      statusBar.registerCommand('testCommand')
-      expect(statusBarItem.command).toBe('testCommand')
+  const assertRender = (nth: number, request: any, type: StatusType) => {
+    const n = nth < 0 ? renderSpy.mock.calls.length - 1 : nth
+    const args = renderSpy.mock.calls[n]
+
+    expect(args[0]).toEqual(request)
+    expect((args[1] as any).type).toEqual(type)
+  }
+
+  describe('register', () => {
+    it('should add 2 commands', () => {
+      statusBar.register(() => undefined)
+      expect(vscode.commands.registerCommand).toBeCalledTimes(2)
+
+      const registerCommand = (vscode.commands.registerCommand as unknown) as jest.Mock<{}>
+      let found = 0
+      registerCommand.mock.calls.forEach(c => {
+        // tslint:disable no-bitwise
+        if (c[0].includes('show-summary-output')) {
+          found |= 0x1
+        }
+        if (c[0].includes('show-active-output')) {
+          found |= 0x2
+        }
+      })
+      expect(found).toEqual(3)
```

Comment:
It is more performant, but is it also easier to read than something like:
```
const calls = registerCommand.mock.calls
expect(calls.some(c => c[0].includes('show-summary-output'))).toBe(true)
expect(calls.some(c => c[0].includes('show-active-output'))).toBe(true)
```
?

Reply:
I like that, especially in tests I agree we should strive for clarity than performance... done